In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

[파생변수 만들기]
메인카페 데이터셋 확정 후 **데이터셋 변경**

In [2]:
cafe2010 = pd.read_csv("./seoul_filtered_cafe_2010이후.csv")

In [3]:
cafe2010 = cafe2010.dropna(subset=['x']) 

In [4]:
cafe2010.info()

<class 'pandas.core.frame.DataFrame'>
Index: 49342 entries, 0 to 49929
Data columns (total 14 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   업소명     49342 non-null  object 
 1   행정동명    49244 non-null  object 
 2   소재지도로명  49342 non-null  object 
 3   소재지지번   49244 non-null  object 
 4   영업자시작일  49342 non-null  object 
 5   소재지시작일  49342 non-null  object 
 6   폐업일자    20879 non-null  object 
 7   업소위치    44775 non-null  object 
 8   총면적     49342 non-null  float64
 9   카페명     49342 non-null  object 
 10  최신시작일   28463 non-null  object 
 11  복층여부    49342 non-null  int64  
 12  x       49342 non-null  float64
 13  y       49342 non-null  float64
dtypes: float64(3), int64(1), object(10)
memory usage: 5.6+ MB


In [5]:
cafe2010.isna().sum()

업소명           0
행정동명         98
소재지도로명        0
소재지지번        98
영업자시작일        0
소재지시작일        0
폐업일자      28463
업소위치       4567
총면적           0
카페명           0
최신시작일     20879
복층여부          0
x             0
y             0
dtype: int64

- 업소명           0
- 행정동명         98  > 
- 소재지도로명        0
- 소재지지번        98 >  
- 영업자시작일        0
- 소재지시작일        0
- 폐업일자      28463 > 결측치는 영업중
- 업소위치       4567 > 모두 지상으로 바꾸거나, 변수 삭제 
- 총면적           0
- 카페명           0
- 최신시작일     20879 > ??
- 복층여부          0
- x             0
- y             0

In [6]:
cafe2010.head()

업소명  행정동명                                     소재지도로명  \
0                  !(느낌)   서교동             서울특별시 마포구 동교로17안길 10, (서교동,1층)   
1                  # (샵)   서교동              서울특별시 마포구 독막로7길 45, (서교동, 3층)   
2  #1 shot in coffee(원샷)  방이1동  서울특별시 송파구 올림픽로 424, 올림픽공원 올림픽수영장 1층 (방이동)   
3                   #113   신림동         서울특별시 관악구 봉천로12길 39, 1층 113호 (신림동)   
4                   #52랩    명동             서울특별시 중구 퇴계로14길 19, 1층 (회현동3가)   

                         소재지지번      영업자시작일      소재지시작일        폐업일자 업소위치  \
0  서울특별시 마포구 서교동  461번지 2호  1층  2010-12-10  2010-12-10  2017-02-07   지상   
1   서울특별시 마포구 서교동  395번지 23호    2013-02-14  2012-01-16  2014-01-28   지상   
2   서울특별시 송파구 방이동  88번지  올림픽공원  2019-04-01  2019-04-01         NaN   지상   
3   서울특별시 관악구 신림동  1426번지 7호    2016-09-12  2015-11-12         NaN   지상   
4  서울특별시 중구 회현동3가  9번지 15호  1층  2016-11-09  2016-11-09  2018-07-20   지상   

      총면적                    카페명       최신시작일  복층여부           x          y  
0   96.70                  !(느낌)         NaN     0  126.916749  37.555387  
1   57.12                  # (샵)         NaN     0  126.919585  37.549846  
2  101.44  #1 shot in coffee(원샷)  2019-04-01     0  127.122959  37.520934  
3   49.50                   #113  2016-09-12     0  126.930207  37.486910  
4   20.27                   #52랩         NaN     0  126.983780  37.559687

In [7]:
df = cafe2010.copy()

# 파생변수 생성1 - 카페 데이터셋
- 카페유형
- 폐업여부
- 총영업일
- 계절
- 월

### 카페유형
- 카페유형 분류
- 행 뽑기 : 고가프차, 저가프차, 중가프차, 개인카페 ('카페유형'으로), 스타벅스('카페명'으로)

In [42]:
df[df['업소명'].str.contains('커피베이', regex=True)].shape

(269, 15)

In [43]:
# 저가 프랜차이즈 - 메가,빽다방,컴포즈,더벤티,매머드
df.loc[df['업소명'].str.contains('메가.*커피|커피.*메가|컴포즈|매머드|메머드|더벤티|빽다방', regex=True), '카페유형'] = '저가 프랜차이즈'

# 고가 프랜차이즈 - 스타벅스,투썸,할리스,파스쿠찌,폴바셋
df.loc[df['업소명'].str.contains('스타벅스|투썸플|할리스|파스쿠찌|파스쿠치|폴바셋|폴 바셋', regex=True), '카페유형'] = '고가 프랜차이즈'

# 중가 프랜차이즈
df.loc[df['업소명'].str.contains('이디야|셀렉토|커피베이|디저트39|디저트 39|바나프레소', regex=True), '카페유형'] = '중가 프랜차이즈'

# 스타벅스만 추출
df.loc[df['업소명'].str.contains('스타벅스', regex=True), '카페명'] = '스타벅스'


In [47]:
# 그 외 카페 > 개인 카페 
df['카페유형'].fillna('개인 카페', inplace=True)

In [48]:
df['카페유형'].isna().sum()

0

In [49]:
df.head()

업소명  행정동명                                     소재지도로명  \
0                  !(느낌)   서교동             서울특별시 마포구 동교로17안길 10, (서교동,1층)   
1                  # (샵)   서교동              서울특별시 마포구 독막로7길 45, (서교동, 3층)   
2  #1 shot in coffee(원샷)  방이1동  서울특별시 송파구 올림픽로 424, 올림픽공원 올림픽수영장 1층 (방이동)   
3                   #113   신림동         서울특별시 관악구 봉천로12길 39, 1층 113호 (신림동)   
4                   #52랩    명동             서울특별시 중구 퇴계로14길 19, 1층 (회현동3가)   

                         소재지지번      영업자시작일      소재지시작일        폐업일자 업소위치  \
0  서울특별시 마포구 서교동  461번지 2호  1층  2010-12-10  2010-12-10  2017-02-07   지상   
1   서울특별시 마포구 서교동  395번지 23호    2013-02-14  2012-01-16  2014-01-28   지상   
2   서울특별시 송파구 방이동  88번지  올림픽공원  2019-04-01  2019-04-01         NaN   지상   
3   서울특별시 관악구 신림동  1426번지 7호    2016-09-12  2015-11-12         NaN   지상   
4  서울특별시 중구 회현동3가  9번지 15호  1층  2016-11-09  2016-11-09  2018-07-20   지상   

      총면적                    카페명       최신시작일  복층여부           x          y  \
0   96.70                  !(느낌)         NaN     0  126.916749  37.555387   
1   57.12                  # (샵)         NaN     0  126.919585  37.549846   
2  101.44  #1 shot in coffee(원샷)  2019-04-01     0  127.122959  37.520934   
3   49.50                   #113  2016-09-12     0  126.930207  37.486910   
4   20.27                   #52랩         NaN     0  126.983780  37.559687   

    카페유형  
0  개인 카페  
1  개인 카페  
2  개인 카페  
3  개인 카페  
4  개인 카페

In [50]:
data1 = df.copy()

#### 행 뽑아서 데이터셋 만들기
- 고가 프차
- 중가 프차
- 저가 프차 
- 개인카페
- 스타벅스

In [62]:
cafe_fren_high = df[df['카페유형']=='고가 프랜차이즈']

cafe_fren_mid = df[df['카페유형']=='중가 프랜차이즈']

cafe_fren_low = df[df['카페유형']=='저가 프랜차이즈']

cafe_nonfren = df[df['카페유형']=='개인 카페']

cafe_starbucks = df[df['카페명']=='스타벅스']

In [64]:
#cafe_fren_high.info() #1724
#cafe_fren_mid.info() #1373
#cafe_fren_low.info() #1848
#cafe_nonfren.info() #44397
cafe_starbucks.info() #691

<class 'pandas.core.frame.DataFrame'>
Index: 691 entries, 354 to 24685
Data columns (total 15 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   업소명     691 non-null    object 
 1   행정동명    690 non-null    object 
 2   소재지도로명  691 non-null    object 
 3   소재지지번   690 non-null    object 
 4   영업자시작일  691 non-null    object 
 5   소재지시작일  691 non-null    object 
 6   폐업일자    41 non-null     object 
 7   업소위치    638 non-null    object 
 8   총면적     691 non-null    float64
 9   카페명     691 non-null    object 
 10  최신시작일   650 non-null    object 
 11  복층여부    691 non-null    int64  
 12  x       691 non-null    float64
 13  y       691 non-null    float64
 14  카페유형    691 non-null    object 
dtypes: float64(3), int64(1), object(11)
memory usage: 86.4+ KB


In [66]:
cafe_fren_high.to_csv('cafe_fren_high.csv' , index=False, encoding='utf-8-sig')
cafe_fren_mid.to_csv('cafe_fren_mid.csv' , index=False, encoding='utf-8-sig')
cafe_fren_low.to_csv('cafe_fren_low.csv' , index=False, encoding='utf-8-sig')
cafe_nonfren.to_csv('cafe_nonfren.csv' , index=False, encoding='utf-8-sig')
cafe_starbucks.to_csv('cafe_starbucks.csv' , index=False, encoding='utf-8-sig')

### 폐업여부

# 파생변수 생성2 - 주변 상권 변수
- 

# 파생변수 생성3 - 지역 특성 변수